### ISSUE TITLE
project status?

### ISSUE THREAD

**Issue by pierreozoux** [whiteout-io/mail-html5#405]
When I see a project with no commit since more than a month, I'm always sad...

Can you provide us the status of your project please?

I hope everything is fine!

Best,

Pierre

**Comment by Xeoncross** [comment]
Lots of great projects don't get a commit for a month. Real-world software sometimes doesn't get updates for months at a time.

I'm not sure how to say this nicely, but It's kind of vain to ask the developers to provide you with a custom update. Who are you again?

**Comment by HLFH** [comment]
> Who are you again?

@pierreozoux is the amazing guy that [empowers users with an ethical hosting solution](https://indiehosters.net/fr_FR/page/home): IndieHosters. If you want to contribute, [don't hesitate to help here.](https://github.com/IndiePaaS/IndiePaaS). [In the French open source community, you have an article](http://framablog.org/2014/12/06/indiehosters-reseau-hebergeurs-qui-vous-libere/) that promotes this open hosting solution in the effort of redecentralizing the Internet.

**Comment by felixhammerl** [comment]
all i can say is that tankred, andris and me are no longer at whiteout, as you can guess from our linkedin profiles and the absence of commits...
please note that i can't speak for whiteout about how the company plans to handle this...

**Comment by pierreozoux** [comment]
@Xeoncross yes, great software don't receive commits anymore. I believe it's been some month `ls` didn't receive any commit, and I still use it every days :)

Thanks @HLFH for the kind presentation, I would have just say, that I am an infrequent user that recommend whiteout as a mail client to people that need crypto.

I was just asking because, in my opinion a mail client should have #373 to be usable on a daily basis.

Thanks @felixhammerl, I think I got it. I hope mailpile will soon provide an easy installer :) Or maybe, I should just try mailveloppe to see if it is recommendable.

@Xeoncross now you got my curiosity, who are you in relation to whiteout?

Will there be some updates on the official website/blog?

**Comment by Xeoncross** [comment]
@pierreozoux I'm nobody. Your ticket title simply sparked my interest. :)

**Comment by HLFH** [comment]
@pierreozoux I know for sure that you support ownCloud hosting with IndieHosters, that ownCloud Mail will get an integration with the CardDAV server in the [0.4 milestone](https://github.com/owncloud/mail/issues/363) and that [RainLoop](https://github.com/RainLoop/rainloop-webmail) webmail (OpenPGP & CardDAV sync support) [fully supports ownCloud 8.2](https://apps.owncloud.com/content/show.php/RainLoop+Webmail?content=165254). Mailpile is still unstable and [does not support CardDAV sync](https://github.com/mailpile/Mailpile/issues/104).

**Comment by pierreozoux** [comment]
@Xeoncross even ls get updated regularly :) http://git.savannah.gnu.org/gitweb/?p=coreutils.git;a=history;f=src/ls.c;h=2ae548d996b236660c6cea10293a31c674c21f01;hb=HEAD

@HLFH yes I'm also following closely ownCloud mail.
I'm not sure about RainLoop, I don't like much their licensing options. As I'm kind of commercial, I don't exactly know what apply to me or not.
I'm also looking forward the new roundCube.

PS: to the actual maintainer of the repo, I think we can close the issue once the situation is clarified on your blog.

**Comment by HLFH** [comment]
@pierreozoux ownCloud is under [AGPLv3 license](https://github.com/owncloud/core/blob/master/COPYING-AGPL). RainLoop is under [AGPLv3 license](https://github.com/RainLoop/rainloop-webmail/blob/master/LICENSE). The ownCloud *Enterprise Edition* is [under *ownCloud Commercial License*](https://owncloud.com/subscriptions/). The RainLoop *Enterprise Edition* is [under *RainLoop Software License](http://www.rainloop.net/purchase/). **AGPLv3 allows Commercial use**. The only issue IMHO is the [background of RainLoop](https://github.com/RainLoop/rainloop-webmail/issues/450) with [AfterLogic WebMail](https://github.com/afterlogic/webmail-lite) but they are now two complete different projects. Some LinuxFR guys were saying that RainLoop is not open source, they were right in the past, but since 6 months, the community edition has now [good licensing options with AGPLv3](https://github.com/RainLoop/rainloop-webmail/issues/8#issuecomment-100014417). You could also [read this FYI](https://github.com/RainLoop/rainloop-webmail/issues/625).

**Comment by pierreozoux** [comment]
Thanks @HLFH for the pointers, it is what I remembered from rainLoop, but indeed now it is better.

**Comment by felixhammerl** [comment]
for what it's worth: i heard that there is an official announcement in the pipeline on the future of the company and this project in particular...

**Comment by felixhammerl** [comment]
FYI: It is official, whiteout as is exists today will cease to exist by the end of the month. which will spark an even more interesting discussion about if and how this client can live on :)

https://blog.whiteout.io/2015/11/30/end-of-life-for-whiteout-key-service-action-required/

**Comment by pierreozoux** [comment]
Thanks for the follow up!

I hope somebody would be interested in continuing your action and continue developing these pieces you created

Thanks for the great work!

(I'll continue yours by hosting free software for people ;) )

**Comment by tanx** [comment]
Tankred from whiteout here. As Felix pointed out the backend services will be shut down soon. Work on the open source client will continue though. We're currently working on Whiteout Mail 2.0 which among other things will use standard HKP servers instead of our proprietary key server.

Others in the community, such as the GPGtools for Mac team, have shown interest in open sourcing our key server and I will try to advocate for this, but can't make that decision alone.

I'll give an update here when there is anything to announce/release.

Thanks,
Tankred

**Comment by tanx** [comment]
An update from my side... Whiteout Mail under it's current name and GitHub Repository will not be continued. We're currently developing a fork named 'hoodiecrow' which is aimed at Gmail users and integrates the RESTful Gmail Apis instead of the full IMAP/SMTP stack (although this decision is not final). You can try it out and find the code here:

https://hoodiecrow.com

**Comment by HLFH** [comment]
It's not a fork. It's a completely different project that does not follow at all the same purposes, at least for the openness.

**Comment by tanx** [comment]
> It's not a fork. It's a completely different project that does not follow at all the same purposes, at least for the openness.

Hence the new name ;)

No but seriously. Most users complained about performance and buggyness of the app. Building a complete IMAP/SMTP stack is really hard, especially with buggy tcp-socket apis (or none at all) depending on the platform. On iOS for instance, the app was so slow that you it was not usable and the App Store reviews were terrible (and rightly so). Rather than have an app that tries to be everything to all people, I'm trying to build something that at least a fraction of the users (Gmail in this case) can and will actually use.

The beauty about open source is though that if you disagree, you're perfectly capable of building your own fork. I just can't recommend using TCP sockets on mobile and would advise against it.

**Comment by HLFH** [comment]
@tanx Thanks for the details and good luck for your project.

**Comment by felixhammerl** [comment]
For those relying on IMAP/SMTP and not using Gmail, I'm in the process of creating an app based on electron, although I'm kinda busy at the moment (new job) and will disclose as soon as there's something to see and try out. But this will take at least a couple of weeks.

**Comment by Xeoncross** [comment]
So do you blame the Cordova TCP plugin implementation for the slow
performance? Other non-Cordova apps seem fine.

**Comment by felixhammerl** [comment]
I'm not saying it's prohibitively slow, but there no way to do STARTTLS properly. That's currently broken and the bug is not being fixed. For 2 years. Neither in Chrome, nor in cca. They just don't care enough as there's probably not enough apps out there leveraging it.

Second, IMAP was not built for mobile. If you're Apple, you can probably make it work, but if you're a 2 person engineering team that builds for multiple platforms ... not so much. IMAP is a stateful protocol with lots of overhead, it was designed for steady (albeit not necessarily fast) connections, not for flaky 2G connections that can break up at any time without you noticing. This can be worked around, but it's a bad starting point. REST is just the better paradigm...

**Comment by tanx** [comment]
Yes! On top of that Cordova on iOS still does not use WKWebView. So no JIT for you!

**Comment by felixhammerl** [comment]
FYI:

https://github.com/MobileChromeApps/mobile-chrome-apps

> The Chrome Apps for Mobile Toolchain is no longer being actively developed. We intend to keep it functional, but do not intend on adding any new features.

**Comment by tanx** [comment]
This was to be expected. The Chrome team's new strategy is clearly on progressive web apps. They explained this at the Chrome Dev Summit. That's why Hoodiecrow is basically just served via GitHub pages. Anyone can fork the repo and host it on their own gh-pages branch.
